tfsdk/attribute.go

```diff
@@ -49,13 +49,25 @@ type Attribute struct {
 	// Optional indicates whether the practitioner can choose not to enter
 	// a value for this attribute or not. Optional and Required cannot both
 	// be true.
+	//
+	// When defining an attribute that has Optional set to true,
+	// and uses PlanModifiers to set a "default value" when none is provided,
+	// Computed must also be set to true. This is necessary because default
+	// values are, in effect, set by the provider (i.e. computed).
 	Optional bool
 
 	// Computed indicates whether the provider may return its own value for
-	// this attribute or not. Required and Computed cannot both be true. If
+	// this Attribute or not. Required and Computed cannot both be true. If
 	// Required and Optional are both false, Computed must be true, and the
 	// attribute will be considered "read only" for the practitioner, with
 	// only the provider able to set its value.
+	//
+	// As mentioned above for Optional, when defining an Attribute
+	// that has "default value", Computed must also be set to true.
+	// Failing to do so, will return the error like:
```

Comment:
Nit: "Above" only makes sense in the context of reading the code or browsing on pkg.go.dev; for folks coming to this documentation via the Go language server, only this field's documentation is shown. I think we can just drop the beginning part of this sentence.

```suggestion
	// When defining an Optional Attribute that has "default value",
	// Computed must also be set to true. Otherwise, Terraform
	// will return an error like:
```